python/lsst/meas/extensions/scarlet/source.py

```diff
@@ -33,9 +33,46 @@
 logger = lsst.log.Log.getLogger("meas.deblender.deblend")
 
 
+def hasEdgeFlux(source, edgeDistance=1):
+    """hasEdgeFlux
+
+    Determine whether or not a source has flux within `edgeDistance`
+    of the edge.
+
+    Parameters
+    ----------
+    source : `scarlet.Component`
+        The source to check for edge flux
+    edgeDistance : int
+        The distance from the edge of the image to consider
+        a source an edge source. For example if `edgeDistance=3`
+        then any source within 3 pixels of the edge will be
+        considered to have edge flux
+
+    Returns
+    -------
+    isEdge: `bool`
+        Whether or not the source has flux on the edge.
+    """
+    assert edgeDistance > 0
+
+    # Use the first band that has a non-zero SED
+    band = np.min(np.where(source.sed > 0)[0])
+    model = source.get_model()[band]
```

Comment:
if you did something like:
```
edges = np.ones(model.shape)
edges[edgeDistance+1:-edgeDistance-1, edgeDistance+1:-edgeDistance-1] = 0
return np.sum(model*edges)>0
```
you could avoid all the ifs, and reduce the number of implicit loops

Reply:
True, but for N = length of a side, that code scales as N^2 whereas the current algorithm scales with N. I verified this in a jupyter notebook.